contracts/protocol/facets/ProtocolInitializationHandlerFacet.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.9;
+
+import "../../domain/BosonConstants.sol";
+import { IBosonProtocolInitializationHandler } from "../../interfaces/handlers/IBosonProtocolInitializationHandler.sol";
+import { ProtocolLib } from "../libs/ProtocolLib.sol";
+import { DisputeResolverHandlerFacet } from "./DisputeResolverHandlerFacet.sol";
+import { TwinHandlerFacet } from "./TwinHandlerFacet.sol";
+import { ProtocolBase } from "../bases/ProtocolBase.sol";
+import { DiamondLib } from "../../diamond/DiamondLib.sol";
+
+/**
+ * @title IBosonProtocolInitializationHandler
+ *
+ * @notice Handle initializion of new versions after 2.1.0.
+ *
+ */
+contract ProtocolInitializationHandlerFacet is IBosonProtocolInitializationHandler, ProtocolBase {
+    /**
+     * @notice Modifier to protect initializer function from being invoked twice for a given version.
+     */
+    modifier onlyUnInitializedVersion(bytes32 _version) {
+        ProtocolLib.ProtocolStatus storage ps = protocolStatus();
+        require(!ps.initializedVersions[_version], ALREADY_INITIALIZED);
+        ps.initializedVersions[_version] = true;
+        _;
+    }
+
+    /**
+     * @notice Initializes the protocol after the deployment.
+     * This function is callable only once for each version
+     *
+     * @param _version - version of the protocol
+     */
+    function initialize(bytes32 _version) public onlyUnInitializedVersion(_version) {
+        bytes32 version = bytes32(bytes("2.2.0"));
+        if (keccak256(abi.encodePacked(_version)) == keccak256(abi.encodePacked(version))) {
```

Comment:
If we pick option 1 then we should revert the transaction if user skips some version. For example, try to initialize 2.3.0 but the last version is 2.1.0.

Reply:
Yes, that's sensible, I agree.

What about first time deployments? Let's say somebody wants to deploy clean v2.5.0. (full suite).
In that case facets can be initialized individually, so all will have correct state. This facet should also be deployed since it stores version information, but it shouldnt do `initV2_5_0` (I consider this `init_X_Y_Z` functions as "diff" init). For that purpuse it should distinguish between first deployment and upgrade.